lib/src/presentation_controller.dart

```diff
@@ -33,31 +47,64 @@ class PresentationController {
 
   void _handleKey(RawKeyEvent value) {
     if (value is RawKeyUpEvent) {
+      print(value.data);
+      if (value.data is RawKeyEventDataMacOs) {
```

Comment:
How about a bit different approach? The main thing we need is `keyCode` from `RawKeyUpEvent`.
How about creating a method that given a `RawKeyEvent` returns us an `int` that is the `keyCode`.
In case that the event cannot be matched, then just return `-1`.
Finally, you would have just a one with a statement that holds all the cases:
```
switch (keyCode) {
      case 20:
      case 21:
      case AndroidKeys.leftKey:
      case AndroidKeys.pageUpKey:
      case AndroidKeys.upKey:
      case MacOSKeys.leftKey:
      case MacOSKeys.pageUpKey:
      case MacOSKeys.upKey:
        _sendAction(PageAction.previous);
        break;
      case 19:
      case 22:
      case AndroidKeys.rightKey:
      case AndroidKeys.pageDownKey:
      case AndroidKeys.downKey:
      case MacOSKeys.rightKey:
      case MacOSKeys.pageDownKey:
      case MacOSKeys.downKey:
        _sendAction(PageAction.next);
        break;
      default:
        break;
    }
```

The only drawback would be that some keys from Android and MacOs would overlap, but for now I do not see that is the case

Reply:
Great ideas! Made the changes already.

For the purposes of the library, there shouldn't be a problem with keycode overlap for now.

Maybe some USB device or presentation remote controller could cause such overlaps but that shouldn't be a concern.